### PR TITLE
madmin: Add listing of transition storage-class

### PR DESCRIPTION
**Pull request by krisis**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added unit tests to cover my changes.
- [ ] I have added/updated functional tests in [mint](https://github.com/minio/mint). (If yes, add `mint` PR # here: )
- [ ] All new and existing tests passed.

### CODE REVIEW

File: cmd/transition-storage-class.go [hunk 1]
```diff
@@ -85,6 +85,38 @@ func (config *TransitionStorageClassConfigMgr) Add(sc madmin.TransitionStorageCl
...
+		storageClasses = append(storageClasses, madmin.TransitionStorageClassConfig{
+			Type: madmin.S3,
+			S3:   &storageClass,
+		})
+	}
+
+	for _, cls := range config.Azure {
+		storageClasses = append(storageClasses, madmin.TransitionStorageClassConfig{
+			Type:  madmin.Azure,
+			Azure: &cls,
```
Comment (poornas) [inline cmd/transition-storage-class.go]: shouldn't you make a local copy here as well?

File: pkg/madmin/storage-class-config.go [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+func (cfg *TransitionStorageClassConfig) Bucket() string {
+	switch cfg.Type {
+	case S3:
+		return cfg.S3.Bucket
+	case Azure:
+		return cfg.Azure.Bucket
+	case GCS:
+		return cfg.Azure.Bucket
+	}
+	panic("unexpected transition storage-class type")
```
Comment (poornas) [inline pkg/madmin/storage-class-config.go]: panic is a bad idea. return error?

File: pkg/madmin/storage-class-config.go [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+func (cfg *TransitionStorageClassConfig) Endpoint() string {
+	switch cfg.Type {
+	case S3:
+		return cfg.S3.Endpoint
+	case Azure:
+		return cfg.Azure.Endpoint
+	case GCS:
+		return cfg.Azure.Endpoint
+	}
+	panic("unexpected transition storage-class type")
```
Comment (poornas) [inline pkg/madmin/storage-class-config.go]: error should be returned rather than panic

File: cmd/transition-storage-class.go (hunk 1 above)
Comment (krisis) [reply cmd/transition-storage-class.go]: yes, thanks for catching this.

**Review by poornas — APPROVED** [review]

**Pull request merged**